src/ARMInterpreter_ALU.cpp

```diff
@@ -321,7 +321,7 @@ A_IMPLEMENT_ALU_OP(AND,_S)
     if (c) cpu->AddCycles_CI(c); else cpu->AddCycles_C(); \
     if (((cpu->CurInstr>>12) & 0xF) == 15) \
     { \
-        cpu->JumpTo(res); \
+        cpu->JumpTo(res & ~3); \
```

Comment:
you only need to clear the lowest bit, the rest is handled in JumpTo

Reply:
Alright let me change that real quick

Reply:
Should be better now